### PR TITLE
[HDInsight]Support compute isolation and return private ip address

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**

The related swagger spec PR is https://github.com/Azure/azure-rest-api-specs/pull/12612

**Testing Guide**

```
az hdinsight create -t spark --version 3.6 -g MyResourceGroup -n MyCluster \
        -p "HttpPassword1234!" --storage-account MyStorageAccount \
        **--enable-compute-isolation** --workernode-size "Standard_E8S_V3" --headnode-size
        "Standard_E8S_V3"
```

`--enable-compute-isolation` is the new added parameter when creating cluster, the above command will create a cluster with compute isolation feature.

**History Notes**

[HDInsight] `az hdinsight create`: Add a new parameter `--enable-compute-isolation` to support create cluster with compute isolation feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by aim-for-better** [comment]
Hi @Juliehzl  @kairu-ms  Could you please help review this PR? Thanks

**Comment by yonzhan** [comment]
@kairu-ms to help with this PR.

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 1]
```diff
@@ -224,6 +224,12 @@ def load_arguments(self, _):
         c.argument('enable_private_link', arg_group='Private Link', arg_type=get_three_state_flag(),
                    help='Indicate whether enable the private link or not.')
 
+        # compute isolation
+        c.argument('enable_compute_isolation', options_list=['--enable-compute-isolation', '--compute-isolation'],
+                   arg_group="Compute Isolation", arg_type=get_three_state_flag(),
+                   help='Indicate whether enable compute isolation or not.')
+        c.argument('host_sku', arg_group='Compute Isolation', help="The dedicated host sku of compute isolation.")
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:231]: Is it an enum argument?

File: src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py [hunk 2]
```diff
@@ -587,3 +598,8 @@ def _with_relay_outbound_and_private_link():
                '--subnet /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers' \
                '/Microsoft.Network/virtualNetworks/fakevnet/subnets/default ' \
                '--resource-provider-connection Outbound --enable-private-link'
+
+    @staticmethod
+    def _with_compute_isolation():
+        return '--version 3.6 -l southcentralus ' \
+               '--enable-compute-isolation --workernode-size Standard_E8S_V3 --headnode-size Standard_E8S_V3'
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py:605]: Can you provide any test with --host-sku argument

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 1 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:231]: > Is it an enum argument?

It is a string.

File: src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py (hunk 2 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py:605]: > Can you provide any test with --host-sku argument

Sure, currently we do not  need to provide this parameter, backend will provide a default value for this.
Anyway, I will update the related test case. Thanks.

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/tests/latest/test_hdinsight_commands.py]: updated.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**